### PR TITLE
modified search metadata to return Astropy.Table

### LINKED ISSUE #3941: Returning Astropy Table instead of Pandas DataFrame when using search_metadata() in JSOCClient.
In JSOCClient when we use search metadata like ->
```python
metadata = client.search_metadata(time, series, wavelength) #this returns pandas dataframe 

print(metadata[['T_OBS', 'WAVELNTH']]) #this filters the result with the columns mentioned
```
I think this inconsistency should be removed because even the ``_lookup_records`` in **jsoc.py** returns Astropy Table when this function is not called by ``search_metadata``. Astropy Tables have matured enough to support same syntax as pandas dataframe to filter the columns.

### PR DESCRIPTION
**Pull request by DShivansh**
Fixes #3941

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -196,6 +196,7 @@ Trivial/Internal Changes
 - Corrected spelling of 'plotting' in timeseries method (changed 'ploting' to 'plotting'). (`#3429 <https://github.com/sunpy/sunpy/pull/3429>`__)
 - Switched to "importlib_metadata" to get package version to speed up import of SunPy. (`#3449 <https://github.com/sunpy/sunpy/pull/3449>`__)
 - Fix tests for `sunpy.data.data_manager` and ensure they are correctly executed with pytest. (`#3550 <https://github.com/sunpy/sunpy/pull/3550>`__)
+- Modified the `search_metadata` function inside `sunpy.net.jsoc` to return the Astropy.Table instead of pandas dataframe. (`#3950 <https://github.com/sunpy/sunpy/pull/3950>`__)
```
Comment (nabobalis) [inline CHANGELOG.rst]: We don't use this changelog directly. You have to add an RST file into the changelogs folder.

**Comment by nabobalis** [comment]
What does the output look like now in terminal compared with before?

**Comment by DShivansh** [comment]
New Output
```
T_REC                  T_OBS          ...       T_REC_epoch      
-------------------- ----------------------- ... -----------------------
2014-01-01T00:00:01Z 2014-01-01T00:00:08.57Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:00:13Z 2014-01-01T00:00:20.58Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:00:25Z 2014-01-01T00:00:32.57Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:00:37Z 2014-01-01T00:00:44.58Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:00:49Z 2014-01-01T00:00:56.57Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:01:01Z 2014-01-01T00:01:08.59Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:01:13Z 2014-01-01T00:01:20.59Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:01:25Z 2014-01-01T00:01:32.57Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:01:37Z 2014-01-01T00:01:44.58Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:01:49Z 2014-01-01T00:01:56.58Z ... 1993.01.01_00:00:04_TAI
2014-01-01T00:02:01Z 2014-01-01T00:02:08.58Z ... 1993.01.01_00:00:04_TAI
```
Old output
```
                                                           T_REC  ...              T_REC_epoch
aia.lev1_euv_12s[2014-01-01T00:00:01Z][304]  2014-01-01T00:00:01Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:13Z][304]  2014-01-01T00:00:13Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:25Z][304]  2014-01-01T00:00:25Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:37Z][304]  2014-01-01T00:00:37Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:49Z][304]  2014-01-01T00:00:49Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:01Z][304]  2014-01-01T00:01:01Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:13Z][304]  2014-01-01T00:01:13Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:25Z][304]  2014-01-01T00:01:25Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:37Z][304]  2014-01-01T00:01:37Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:49Z][304]  2014-01-01T00:01:49Z  ...  1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:02:01Z][304]  2014-01-01T00:02:01Z  ...  1993.01.01_00:00:04_TAI

[11 rows x 176 columns]
```

File: CHANGELOG.rst (hunk 1 above)
Comment (DShivansh) [reply CHANGELOG.rst]: do I have to remove this change from here?

**Comment by nabobalis** [comment]
Looking at that, I do think we should keep the first column from the old version.

**Comment by DShivansh** [comment]
first column is actually the index value, should I create a column named index in the astropy.Table and put index-values in it and do the indexing according to them?

**Comment by nabobalis** [comment]
Sure.

File: CHANGELOG.rst (hunk 1 above)
Comment (DShivansh) [reply CHANGELOG.rst]: @nabobalis should I remove the change in CHANGELOG and only upload a file in changes.rst? or do we have to keep both?

Comment (nabobalis) [reply CHANGELOG.rst]: Remove this change and add a changelog file in to the changelog folder.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -198,6 +198,7 @@ Trivial/Internal Changes
 - Fix tests for `sunpy.data.data_manager` and ensure they are correctly executed with pytest. (`#3550 <https://github.com/sunpy/sunpy/pull/3550>`__)
 
 
+
```
Comment (nabobalis) [inline CHANGELOG.rst]:
```suggestion
```

File: changelog/3950.trivial.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Modified the `search_metadata` function inside `sunpy.net.jsoc` to return the Astropy.Table instead of pandas dataframe.
```
Comment (nabobalis) [inline changelog/3950.trivial.rst]: This change is technically breaking, so it should have it's type changed from rivial.

**Comment by nabobalis** [comment]
What does the output look like now?

File: changelog/3950.trivial.rst (hunk 3 above)
Comment (DShivansh) [reply changelog/3950.trivial.rst]: ok

**Comment by DShivansh** [comment]
Output
```python
 Index                    ...       T_REC_epoch      
------------------------------------------- ... -----------------------
aia.lev1_euv_12s[2014-01-01T00:00:01Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:13Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:25Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:37Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:00:49Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:01Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:13Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:25Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:37Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:01:49Z][304] ... 1993.01.01_00:00:04_TAI
aia.lev1_euv_12s[2014-01-01T00:02:01Z][304] ... 1993.01.01_00:00:04_TAI
```

**Comment by nabobalis** [comment]
As things stand, this PR is good to go but we need to resolve how to deal with the breaking change or even if we need this function anymore. 

These two questions are unresolved right now and as such, we can't merge this PR until then.

**Comment by DShivansh** [comment]
@nabobalis should we raise some type of warning that indicates the user that we don't return pandas dataframe anymore instead it is returning the astropy df.

**Comment by nabobalis** [comment]
The warning needs to have a way to turn it off and warning the user as they run the code after a update isn't really an advanced warning.

**Comment by DShivansh** [comment]
can we use some kind of flag in sunpy rc to turn off the warning once user knows about it?

**Comment by nabobalis** [comment]
I don't have a good solution right now.

**Comment by dstansby** [comment]
The way I have seen this sort of deprecation done previously is to add a `return_astropy_table` keyword argument, which defaults to `None`. If it is `None`, the code warns, telling the user can do `return_astropy_table=True` to silence the error, and if it is `False` it still uses the old behavior, but raises a warning saying that this functionality will be removed in a future version.

**Comment by nabobalis** [comment]
I think with the current GSoC project, we will probably remove this method as a result I will be closing this.

I want to thank @DShivansh for opening the PR and I am sorry we decided to change the goal posts.

**Pull request closed without merging**